awx/main/tasks.py

```diff
@@ -372,16 +391,29 @@ def gather_analytics():
             if acquired is False:
                 logger.debug('Not gathering analytics, another task holds lock')
                 return
-            try:
-                tgz = analytics.gather()
-                if not tgz:
-                    return
-                logger.info('gathered analytics: {}'.format(tgz))
-                analytics.ship(tgz)
-                settings.AUTOMATION_ANALYTICS_LAST_GATHER = gather_time
-            finally:
-                if os.path.exists(tgz):
-                    os.remove(tgz)
+            subset = list(all_collectors().keys())
+            incremental_collectors = []
+            for collector in expensive_collectors():
+                if collector in subset:
+                    subset.remove(collector)
+                    incremental_collectors.append(collector)
+
+            # Cap gathering at 4 weeks of data if there has been no data gathering
+            since = last_time or (gather_time - timedelta(weeks=4))
```

Comment:
@wenottingham we wanted to go over 4 weeks in https://github.com/RedHatInsights/tower-analytics-backend/issues/97 ?

Is this the right time? :-)

Reply:
We'd need to use different timestamping/tracking then.

Reply:
Resolving this, we'd need a different way to track what we've gathered and sent to do most-recent-first. Not saying we can't, but don't want to do that now.